README.md

```diff
@@ -62,6 +62,14 @@ You can also pass the name of your project to the tool using the `--name` or `-n
 cargo generate --git https://github.com/githubusername/mytemplate.git --name myproject
 ```
 
+If the git repository contains multiple templates, the specific subfolder in the git repository may be specified like this:
+
+```sh
+cargo generate --git "https://github.com/githubusername/mytemplate.git#<relative-template-path>"
```

Comment:
note: TBH I'm not yet fully convinced by this "special" URL notation. 

I'm in favour of keep the git URL a regular URL and better add an unnamed argument to keep the folder in there separated.

I know the that this would conflict with the <favorite> argument but this can be solved by a bit of logic:
1. <favorite> is only allowed in disjunction with `-g | --git`
2. so if <favourite> is present then <relative-template-path> would be the 2nd unnamed argument and optional
3. if `-g | --git` is present then the first named argument is taken as <relative-template-path> and not as it is now as <favourite>

Reply:
Hmm, this might actually be doable - so, if `-g` | `--git` argument is specified, we use the unnamed argument (normally used to specify a 'favorite') as the internal folder? I like that 😃 

In this case, when a folder needs to be specified in a favorite, we need to name the argument though... any idea?

Reply:
hm, good point. I forgot about the configuration file structure and that the `<subfolder>` would make sense too.

I suggest to add a named optional parameter in the favourites config with name `subfolder`.

like:

```toml
[favorites.cli]
description = "CLI template for a rust binary"
git = "https://github.com/sassman/cli-template-rs.git"
branch = "master"
subfolder = "examples/demo"
```

so this `subfolder = "examples/demo"` is then the new part.

Reply:
Regarding name resolution, if the user does not specify a `--name`, should we use the subfolder name (if specified) instead of the repo name?

I'm thinking yes!